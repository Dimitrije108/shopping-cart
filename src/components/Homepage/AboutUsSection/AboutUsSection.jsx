import styles from "./AboutUsSection.module.css";
import CEC from "/src/assets/icons/Corellian_Engineering_Corporation.svg";

export default function AboutUsSection() {
	return (
		<section className={styles.aboutUsSection}>
			<div className={styles.aboutUsWrapper}>
				<h2>About Us</h2>
				<div className={styles.aboutUsText}>
					<p>At <span className={styles.bold}>Shipyard</span>, as part of the <span className={styles.bold}>Corellian Engineering Corporation (CEC)</span>, we’ve been at the forefront of starship innovation for generations. Known across the galaxy for producing some of the most reliable and customizable vessels, our ships are designed for traders, adventurers, and defense forces alike.</p>
					<p>Whether you're smuggling spice through Hutt space, running high-priority cargo for the Republic, or just looking to outrun Imperial entanglements, we’ve got the perfect ship for you.
						From the legendary <span className={styles.bold}>YT-series freighters</span>, including the YT-1300, to the battle-tested <span className={styles.bold}>CR90 corvettes</span>, our catalog features ships engineered for performance, endurance, and adaptability. Whether you’re a privateer in need of a durable transport or a fleet commander seeking tactical superiority, CEC offers unparalleled craftsmanship and modularity to suit every need.</p>
					<p>Every ship we build carries the spirit of <span className={styles.bold}>Corellian ingenuity</span>—fast, tough, and endlessly upgradeable. Explore our selection today and see why <span className={styles.bold}>Shipyard</span> is the top choice for independent pilots and galactic fleets alike.</p>
				</div>
				<div className={styles.aboutUsLogo}>
					<h3 title="Shipyard">Shipyard</h3>
					<img 
						src={CEC}
						alt="Corellia Engineering Corporation logo" 
						width={120}
						title="Corellia Engineering Corporation"
						className={styles.cecLogo}
					/>
				</div>
			</div>
		</section>
	)
};
