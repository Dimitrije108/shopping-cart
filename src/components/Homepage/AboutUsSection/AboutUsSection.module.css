.aboutUsSection {
	min-height: calc(100vh - 60px);

	background-image: url("/src/assets/images/dune-night-sand-death-star.jpg");
	background-size: cover;
	background-repeat: no-repeat;
	background-position: center;

	display: flex;
	justify-content: center;
}

.aboutUsWrapper {
	margin: 1rem;
	max-width: 1250px;
	height: 100%;

	text-align: center;

	display: flex;
	flex-direction: column;
	justify-content: space-around;
}

.aboutUsWrapper > h2 {
	margin: 1rem;
	font-family: Trench;
	font-size: 3rem;
}

.aboutUsText {
	max-width: 75ch;
	padding: 1rem;

	font-size: 1.1rem;
	background-color: rgba(0, 0, 0, 0.4);
	text-align: start;
}

.aboutUsText > p {
	margin: 1rem 0;
}

.bold {
	font-family: 'DINNext bold';
}

.aboutUsLogo {
	margin: 1rem;
}

.aboutUsLogo > h3 {
	font-family: Imperial;
	font-size: 3rem;
}

@media (max-width: 720px) {
	.aboutUsSection {
		background-image: url("/src/assets/images/dune-night-sand-death-star-cropped.jpg");
		background-size: cover;
		background-repeat: no-repeat;
		background-position: center;
	}
}
