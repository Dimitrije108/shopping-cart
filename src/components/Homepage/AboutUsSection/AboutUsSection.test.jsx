import { describe, it, expect } from "vitest";
import { render, screen  } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AboutUsSection from "./AboutUsSection";

describe("Homepage's About Us Section component", () => {
	it("About Us Section snapshot", () => {
		const aboutUsSection = render(
			<MemoryRouter>
				<AboutUsSection />
			</MemoryRouter>
		)

		expect(aboutUsSection).toMatchSnapshot();
	});

	it("About Us Section elements render correctly", () => {
		render(
			<MemoryRouter>
				<AboutUsSection />
			</MemoryRouter>
		)
		
		const heading = screen.getByText("About Us");
		const paragraphs = screen.getAllByRole("paragraph");
		const companyName = screen.getByTitle("Shipyard");
		const cecLogo = screen.getByTitle("Corellia Engineering Corporation");
		
		expect(heading).toBeInTheDocument();
		expect(paragraphs).toHaveLength(3);
		expect(companyName).toHaveTextContent("Shipyard");
		expect(cecLogo).toBeInTheDocument();
		expect(cecLogo).toHaveAttribute("src", "/src/assets/icons/Corellian_Engineering_Corporation.svg");
	});
});
