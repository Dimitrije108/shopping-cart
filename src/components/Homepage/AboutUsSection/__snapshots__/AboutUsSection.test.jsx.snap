// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage's About Us Section component > About Us Section snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="_aboutUsSection_bf1662"
      >
        <div
          class="_aboutUsWrapper_bf1662"
        >
          <h2>
            About Us
          </h2>
          <div
            class="_aboutUsText_bf1662"
          >
            <p>
              At 
              <span
                class="_bold_bf1662"
              >
                Shipyard
              </span>
              , as part of the 
              <span
                class="_bold_bf1662"
              >
                Corellian Engineering Corporation (CEC)
              </span>
              , we’ve been at the forefront of starship innovation for generations. Known across the galaxy for producing some of the most reliable and customizable vessels, our ships are designed for traders, adventurers, and defense forces alike.
            </p>
            <p>
              Whether you're smuggling spice through Hutt space, running high-priority cargo for the Republic, or just looking to outrun Imperial entanglements, we’ve got the perfect ship for you. From the legendary 
              <span
                class="_bold_bf1662"
              >
                YT-series freighters
              </span>
              , including the YT-1300, to the battle-tested 
              <span
                class="_bold_bf1662"
              >
                CR90 corvettes
              </span>
              , our catalog features ships engineered for performance, endurance, and adaptability. Whether you’re a privateer in need of a durable transport or a fleet commander seeking tactical superiority, CEC offers unparalleled craftsmanship and modularity to suit every need.
            </p>
            <p>
              Every ship we build carries the spirit of 
              <span
                class="_bold_bf1662"
              >
                Corellian ingenuity
              </span>
              —fast, tough, and endlessly upgradeable. Explore our selection today and see why 
              <span
                class="_bold_bf1662"
              >
                Shipyard
              </span>
               is the top choice for independent pilots and galactic fleets alike.
            </p>
          </div>
          <div
            class="_aboutUsLogo_bf1662"
          >
            <h3
              title="Shipyard"
            >
              Shipyard
            </h3>
            <img
              alt="Corellia Engineering Corporation logo"
              class="_cecLogo_bf1662"
              src="/src/assets/Corellian_Engineering_Corporation.svg"
              title="Corellia Engineering Corporation"
              width="120"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="_aboutUsSection_bf1662"
    >
      <div
        class="_aboutUsWrapper_bf1662"
      >
        <h2>
          About Us
        </h2>
        <div
          class="_aboutUsText_bf1662"
        >
          <p>
            At 
            <span
              class="_bold_bf1662"
            >
              Shipyard
            </span>
            , as part of the 
            <span
              class="_bold_bf1662"
            >
              Corellian Engineering Corporation (CEC)
            </span>
            , we’ve been at the forefront of starship innovation for generations. Known across the galaxy for producing some of the most reliable and customizable vessels, our ships are designed for traders, adventurers, and defense forces alike.
          </p>
          <p>
            Whether you're smuggling spice through Hutt space, running high-priority cargo for the Republic, or just looking to outrun Imperial entanglements, we’ve got the perfect ship for you. From the legendary 
            <span
              class="_bold_bf1662"
            >
              YT-series freighters
            </span>
            , including the YT-1300, to the battle-tested 
            <span
              class="_bold_bf1662"
            >
              CR90 corvettes
            </span>
            , our catalog features ships engineered for performance, endurance, and adaptability. Whether you’re a privateer in need of a durable transport or a fleet commander seeking tactical superiority, CEC offers unparalleled craftsmanship and modularity to suit every need.
          </p>
          <p>
            Every ship we build carries the spirit of 
            <span
              class="_bold_bf1662"
            >
              Corellian ingenuity
            </span>
            —fast, tough, and endlessly upgradeable. Explore our selection today and see why 
            <span
              class="_bold_bf1662"
            >
              Shipyard
            </span>
             is the top choice for independent pilots and galactic fleets alike.
          </p>
        </div>
        <div
          class="_aboutUsLogo_bf1662"
        >
          <h3
            title="Shipyard"
          >
            Shipyard
          </h3>
          <img
            alt="Corellia Engineering Corporation logo"
            class="_cecLogo_bf1662"
            src="/src/assets/Corellian_Engineering_Corporation.svg"
            title="Corellia Engineering Corporation"
            width="120"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
